[topic-gpio] samples: led_apa102c_bitbang: Convert to new GPIO API

carlescufi opened the pull request:
Conver to the new GPIO API, using raw access since there is no DT
definitions for this particular usage of the pins.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

carlescufi committed:
samples: led_apa102c_bitbang: Convert to new GPIO API

Conver to the new GPIO API, using raw access since there is no DT
definitions for this particular usage of the pins.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

pabigot commented on samples/drivers/led_apa102c_bitbang/src/main.c:
```diff
@@ -79,12 +79,12 @@ void main(void)
 
 	/* Setup GPIO output */
-	ret = gpio_pin_configure(gpio_dev, GPIO_DATA_PIN, (GPIO_DIR_OUT));
+	ret = gpio_pin_configure(gpio_dev, GPIO_DATA_PIN, (GPIO_OUTPUT));
```
There's really no reason why that parameter needs to be in parentheses, but changing it isn't necessary.

carlescufi replied:
Fixed, thanks

carlescufi commented:
Untested, I don't have the hardware.

pabigot commented on samples/drivers/led_apa102c_bitbang/src/main.c:
```diff
@@ -55,11 +55,11 @@ void send_rgb(struct device *gpio_dev, u32_t rgb)
 	for (i = 0; i < 32; i++) {
 		/* MSB goes in first */
-		gpio_pin_write(gpio_dev, GPIO_DATA_PIN, !!(rgb & 0x80000000));
+		gpio_pin_set_raw(gpio_dev, GPIO_DATA_PIN, !!(rgb & 0x80000000));
```
MISRA-inclined folks would prefer this to change to `(rgb & 0x80000000U) != 0` or (better) `(rgb & BIT(31)) != 0`.

carlescufi replied:
Why not, since we are touching this file. Fixed, thanks.

pabigot reviewed (approved):
Possible cleanups but this should be fine.

carlescufi committed:
samples: led_apa102c_bitbang: Clean conditional expression

Clean the expression so that it uses the BIT() macro and doesn't abuse
the not (!) operator.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>
